src/AppInstallerCLICore/Commands/PinCommand.cpp

```diff
@@ -185,9 +199,22 @@ namespace AppInstaller::CLI
 
     void PinRemoveCommand::ExecuteInternal(Execution::Context& context) const
     {
+        if (context.Args.Contains(Execution::Args::Type::Id))
+        {
+            // When we are given an ID, just un-pin that available package without checking for installed.
+            // This helps when there are matching issues, for example due to multiple side-by-side installs.
+            context <<
+                Workflow::OpenSource();
+        }
+        else
+        {
+            // If not working from just ID, try matching a single installed package
+            context <<
+                Workflow::OpenSource() <<
+                Workflow::OpenCompositeSource(Repository::PredefinedSource::Installed);
+        }
+

```

Comment:
This section is repeated between PinAdd and PinRemove. Perhaps it would make sense to have a workflow task like `OpenSourceForPinning`, or even abstract it to something like `OpenSourceFor( OperationType::Pin)` so that the source opening for list/update/show/install could be brought back into a single workflow task.